### PR TITLE
Converged aie rt

### PR DESCRIPTION
**Pull request by eddierichter-amd**
These changes require the modification to libxaie that allows libxaie calls to be communicate via the AIR driver to the command processor on the card. Want to merge it into Steve's aie-rt so all of our modifications can exist in a single repo.

### CODE REVIEW

**Comment by makslevental** [comment]
@stephenneuendorffer is this good to go?

**Comment by eddierichter-amd** [comment]
> @stephenneuendorffer is this good to go?

Steve can answer but from my point of view the only thing missing is making sure these changes don't mess up the RyzenAI flow, I have already tested that this convergence doesn't mess up the VCK5000 flow. I am getting a box setup right now so I can test it when that is complete, if you can help expedite and run the flow on your Ryzen AI box using this fork of aie-rt that would be awesome!

**Comment by makslevental** [comment]
> Steve can answer but from my point of view the only thing missing is making sure these changes don't mess up the RyzenAI flow, I have already tested that this convergence doesn't mess up the VCK5000 flow. I am getting a box setup right now so I can test it when that is complete, if you can help expedite and run the flow on your Ryzen AI box using this fork of aie-rt that would be awesome!

What is the flow in question? Something at the air level I guess?

**Comment by eddierichter-amd** [comment]
> > Steve can answer but from my point of view the only thing missing is making sure these changes don't mess up the RyzenAI flow, I have already tested that this convergence doesn't mess up the VCK5000 flow. I am getting a box setup right now so I can test it when that is complete, if you can help expedite and run the flow on your Ryzen AI box using this fork of aie-rt that would be awesome!
> 
> What is the flow in question? Something at the air level I guess?

No all at mlir-aie. Really just changing your branch of mlir-aie to point to my fork of aie-rt and making sure that you can build mlir-aie, build a test, and run it on the board.

**Comment by makslevental** [comment]
> No all at mlir-aie. Really just changing your branch of mlir-aie to point to my fork of aie-rt and making sure that you can build mlir-aie, build a test, and run it on the board.

There's already a ryzen runner there and your PR passes all tests on it: https://github.com/Xilinx/mlir-aie/actions/runs/7522856029. I can double check locally but I'm configured almost exactly the same as that runner.

**Comment by eddierichter-amd** [comment]
Another thing is, once we merge this, can we change the branch name to something like `experimental-aie-rt` or something that doesn't have Phoenix in the name?

**Comment by eddierichter-amd** [comment]
Alright this was breaking the Phoenix flow but now I can test it and it is working on Phoenix. @stephenneuendorffer Let me know if you think we need anything else before merging.

File: driver/src/io_backend/ext/xaie_amdair.c [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+*
+* @return	None.
+*
+* @note		The global IO instance is a singleton and freed when
+* the reference count reaches a zero.
+*
+*******************************************************************************/
+static AieRC XAie_AmdairIO_Finish(void *IOInst)
+{
+	XAie_AmdairIO *DevInst = (XAie_AmdairIO *)IOInst;
```
Comment (makslevental) [inline driver/src/io_backend/ext/xaie_amdair.c]: nit: `Amdair` -> `AmdAir`

File: driver/src/io_backend/ext/xaie_amdair.c [hunk 2]
```diff
@@ -0,0 +1,615 @@
...
+	// recover sysfs path
+	if (!DevInst->IOInst) {
+		XAIE_ERROR("sysfs path not present!\n");
+		return XAIE_ERR;
+	}
+
+	IOInst->BaseAddr = DevInst->BaseAddr;
+	IOInst->NpiBaseAddr = XAIE_NPI_BASEADDR;
+
+	IOInst->AmdairAddress= malloc(strlen((const char *)DevInst->IOInst) + strlen("/address") + 1);
```
Comment (makslevental) [inline driver/src/io_backend/ext/xaie_amdair.c]: nit: spacing around the `=` here.

File: driver/src/io_backend/ext/xaie_amdair.c [hunk 3]
```diff
@@ -0,0 +1,615 @@
...
+*
+*******************************************************************************/
+static AieRC XAie_AmdairIO_MaskWrite32(void *IOInst, u64 RegOff, u32 Mask,
+		u32 Value)
+{
+	u32 RegVal;
+
+	AieRC rc = XAie_AmdairIO_Read32(IOInst, RegOff, &RegVal);
+	if (rc != XAIE_OK)
+		return rc;
```
Comment (makslevental) [inline driver/src/io_backend/ext/xaie_amdair.c:207]: nit: `XAIE_ERROR` here.

Comment (eddierichter-amd) [reply driver/src/io_backend/ext/xaie_amdair.c:207]: Addressed in [6de16c1](https://github.com/stephenneuendorffer/aie-rt/pull/4/commits/6de16c1da98d0def29a457547e30e8c1a6f0bb43)

File: driver/src/io_backend/ext/xaie_amdair.c (hunk 2 above)
Comment (eddierichter-amd) [reply driver/src/io_backend/ext/xaie_amdair.c]: Addressed in [6de16c1](https://github.com/stephenneuendorffer/aie-rt/pull/4/commits/6de16c1da98d0def29a457547e30e8c1a6f0bb43)

File: driver/src/io_backend/ext/xaie_amdair.c (hunk 1 above)
Comment (eddierichter-amd) [reply driver/src/io_backend/ext/xaie_amdair.c]: Addressed in [6de16c1](https://github.com/stephenneuendorffer/aie-rt/pull/4/commits/6de16c1da98d0def29a457547e30e8c1a6f0bb43)

**Comment by makslevental** [comment]
Next is a corresponding bump in mlir-aie?

**Comment by eddierichter-amd** [comment]
> Next is a corresponding bump in mlir-aie?

Yeah, I just bumped it in my PR of mlir-aie (https://github.com/Xilinx/mlir-aie/pull/570) which we should be good to go on once the CI runs.

**Pull request merged**